daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs

```diff
@@ -186,7 +192,7 @@ generateDalfRule =
         unsimplifiedRawDalf <- use_ GenerateRawDalf file
         let rawDalf = LF.simplifyModule unsimplifiedRawDalf
         setPriority PriorityGenerateDalf
-        pure $ toIdeResult $ do
+        pure $ either (, Nothing) (([],) . Just) $ do
```

Comment:
Not sure duplicating this logic is an improvement.

Reply:
I'm not convinced this pattern should exist anywhere - it should really be fused with whatever is producing it. By making it internal we can do that more easily. I suspect there's a further cleanup to be done here, but was aiming for one step at a time.

Reply:
Having operations that can only emit errors but not warnings doesn’t seem that unreasonable to me but I don’t feel particularly strongly about this.

Reply:
OK, I think you've convinced me. I put that bit back. Not sure `toIdeResult` is the right name, but we can solve that in future.

Reply:
I full agree that `toIdeResult` is a terrible name, I was actually trying to change it recently but failed to come up with something better.